Test/Models_Functions/MP2RAGE/mp2rage_blochsim_test.m

```diff
@@ -0,0 +1,51 @@
+clc; clear; close all;
+
+alpha1 = deg2rad(5);  % Flip angle for first VFA block (radians)
+alpha2 = deg2rad(5);  % Flip angle for second VFA block (radians)
+N_VFA1 = 50;         % Number of excitations in first VFA block
+N_VFA2 = 50;         % Number of excitations in second VFA block
+TI1 = 700;            % First inversion time (ms)
+TI2 = 2500;           % Second inversion time (ms)
+T1 = 1000;            % Longitudinal relaxation time (ms)
+T2 = 80;              % Transverse relaxation time (ms)
+TE = 5;               % Echo time (ms)
+TR = 5000;            % Repetition time (ms)
+df = 0;               % Off-resonance frequency (Hz)
+inc = 0;              % RF phase increment (degrees)
+
+[MLong_TI1, MLong_TI2, Msig1, Msig2] = mp2rage_blochsim(alpha1, alpha2, N_VFA1, N_VFA2, TI1, TI2, T1, T2, TE, TR, df, inc);
```

Comment:
I tried this function for the following parameters, and got the following outputs for each VFA bloch. The second one seems kinda odd for me for the 15 degrees, it's possible though I just haven't seen that before.

<img width="558" alt="Screenshot 2025-02-03 at 10 53 57 AM" src="https://github.com/user-attachments/assets/aae37bec-ee7d-4c5f-9584-ee6bb0bb731c" />

<img width="955" alt="Screenshot 2025-02-03 at 10 52 09 AM" src="https://github.com/user-attachments/assets/0c55ea38-39d6-4e1b-845f-dc9104af41d0" />

Reply:
For these same parameters, I looped through alpha1 values of 1 to 90 degrees and saved the last measurement value for TI1, then plotted it. It looks like the expected trend - great!

<img width="761" alt="Screenshot 2025-02-03 at 10 58 48 AM" src="https://github.com/user-attachments/assets/60d113e9-e664-49a5-a85d-106ef28edc4f" />

Reply:
Same thing but with these parameters to test IR (only one VFA excitation, TI values ranging, T1 = 1s, etc), the trend looks great and the nulling time is exactly the same as what my MOOC plot shows it should be - excellent!
<img width="683" alt="Screenshot 2025-02-03 at 11 11 55 AM" src="https://github.com/user-attachments/assets/f0427e83-16e8-4ddf-b159-3b8910e5ac4f" />
<img width="784" alt="Screenshot 2025-02-03 at 11 11 44 AM" src="https://github.com/user-attachments/assets/d9496148-69c8-4cb0-b88b-0062fcd3b2e9" />
<img width="812" alt="Screenshot 2025-02-03 at 11 01 53 AM" src="https://github.com/user-attachments/assets/f67bf613-8bc5-4df0-a4d2-749ef688de82" />

Reply:
So these are the kind of tests I think you should do; maybe formalize them, if you don't know how to write tests using the matlab test suite I can help you out with that, I mocke up some simple examples for an ISMRM tutorial a few years back, here are the MATLAB examples for a simple addition function: https://github.com/ismrm-coding-secret-session/testing-your-code/tree/matlab , and here were my slides: https://github.com/ismrm-coding-secret-session/slides/blob/master/pdfs/testing.pdf